File: cuaderno_mantenimiento_flutter/lib/infrastructure/dtos/create-client-dto.dart
class CreateClientDto {
  final String nombre;
  final String correo;
  final String password;
  final String domicilio;
  final String localidad;
  final String provincia;
  final String codigoPostal;
  final String telefono;

  CreateClientDto({
    required this.nombre,
    required this.correo,
    required this.password,
    required this.domicilio,
    required this.localidad,
    required this.provincia,
    required this.codigoPostal,
    required this.telefono,
  });

  // Puedes agregar un método para convertirlo a JSON
  Map<String, dynamic> toJson() {
    return {
      'nombre': nombre,
      'correo': correo,
      'password': password,
      'domicilio': domicilio,
      'localidad': localidad,
      'provincia': provincia,
      'codigo_postal': codigoPostal,
      'telefono': telefono,
    };
  }}

File: cuaderno_mantenimiento_backend/src/client/dto/create-client.dto.ts
import { IsString, IsUUID, IsNotEmpty, IsArray, IsOptional, IsEmail, IsIn, Matches } from 'class-validator';
import { CreatePersonDto } from 'src/person/dto/create-person.dto'; // Asegúrate de que el DTO de Persona esté correctamente importado


export class CreateClientDto {
 @IsString()
  nombre: string;

  @IsEmail()
  correo: string;

  @IsString()
  password: string;



  @IsNotEmpty()
  @IsString()
  domicilio: string;

  @IsNotEmpty()
  @IsString()
  localidad: string;

  @IsNotEmpty()
  @IsString()
  provincia: string;

  @IsNotEmpty()
  @IsString()
  @Matches(/^\d{5}$/, { message: 'El código postal debe tener 5 dígitos.' })
  codigo_postal: string;
  

  @IsNotEmpty()
  @IsString()
  @Matches(/^\d{9}$/, { message: 'El teléfono debe tener 9 dígitos.' })
  telefono: string;
  

  
  
}
